### PR TITLE
Create HOME menu

### LINKED ISSUE #1: TV Home menu
As a user,
I want to be able to see following options in menu:
- Internet Contents
- Digital Programme List
- Digital EPG
- Applications
- Recordings
- Media
- Settings,
so that user can see all options in menu.

### PR DESCRIPTION
**Pull request by Varga4x4**

### CODE REVIEW

File: index.css [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+#app {
+    border: 2px solid green;
+    height: 600px;
+    width: 900px;
+    display: flex;
+    justify-content: flex-start;
+    align-items: center;
+}
+
+main {
```
Comment (grandalf6) [inline index.css:23]: controlling height is never great - you could use padding on parent component instead

File: index.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
+const HOME_MENU_BUTTONS_ARRAY = [
+    {
+        id: "internet-contents",
+        className: "home-options",
```
Comment (grandalf6) [inline index.js]: if className is always the same, then there is no need to specify it with each element

File: index.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+// RENDER_HOME_MENU
+const renderHomeMenu = () => {
+    const mainElement = createElement("main", undefined, appElement)
+
+    createElement("p", {
+        innerText: "HOME"
+    }, mainElement)
+
+    HOME_MENU_BUTTONS_ARRAY.forEach((option) => {
+        const homeMenuOptionsButtonElement = createElement("button", {
```
Comment (grandalf6) [inline index.js]: looks like this variable is not used

File: index.js [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+            id: option.id,
+            className: option.className,
```
Comment (grandalf6) [inline index.js]: here instead of passing option.className you could use string as class name is always the same

File: index.js [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+// RENDER_HOME_MENU
+const renderHomeMenu = () => {
+    const mainElement = createElement("main", undefined, appElement)
+
+    createElement("p", {
+        innerText: "HOME"
+    }, mainElement)
+
+    HOME_MENU_BUTTONS_ARRAY.forEach((option) => {
```
Comment (grandalf6) [inline index.js]: here instead of options you could write {id, className, innerText}

Comment (grandalf6) [reply index.js]: It is called destructurisation

File: index.js [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+// HELPERS
+const createElement = (tagType, attrObj, parentElement) => {
+    if (!tagType) {
+        throw new Error("Error 2");
+    }
+
+    const newElement = document.createElement(tagType)
+
+    if (attrObj) {
+        Object.entries(attrObj).forEach((entry) => {
```
Comment (grandalf6) [inline index.js]: here you could descructurize entry

Comment (grandalf6) [reply index.js]: and you don't need {} in the end of line

File: index.js (hunk 5 above)
Comment (grandalf6) [reply index.js]: also, if you get rid of variable you don't need in the body of this arrow function, then {} and the end of this line will be redundant

File: index.js [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+        innerText: "Media"
+    },
+    {
+        id: "settings",
+        className: "home-options",
+        innerText: "Settings"
+    },
+]
+
+const ELEMENT_IDS = {
```
Comment (grandalf6) [inline index.js]: Are you using this object?

**Review by grandalf6 — CHANGES_REQUESTED** [review]
Almost great - just small syntax issues

File: index.css (hunk 1 above)
Comment (Varga4x4) [reply index.css:23]: In an other way but solved

File: index.js (hunk 2 above)
Comment (Varga4x4) [reply index.js]: Solved

File: index.js (hunk 3 above)
Comment (Varga4x4) [reply index.js]: Solved

File: index.js (hunk 4 above)
Comment (Varga4x4) [reply index.js]: Solved

File: index.js (hunk 5 above)
Comment (Varga4x4) [reply index.js]: I read about destructuring but not clear how to use it here.

File: index.js (hunk 6 above)
Comment (Varga4x4) [reply index.js]: Same

File: index.js (hunk 7 above)
Comment (Varga4x4) [reply index.js]: No, I was thinking to use it in the future with the sub menu.

File: index.css [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+    border: 2px solid blue;
+    display: flex;
+    flex-direction: column;
+    font-size: 30px;
+    gap: 20px;
+    margin: 0;
```
Comment (grandalf6) [inline index.css]: Do you need margin: 0?

File: index.js (hunk 7 above)
Comment (grandalf6) [reply index.js]: Ok, so you have ids multiplied in these both arrays

File: index.js (hunk 6 above)
Comment (grandalf6) [reply index.js]: ```Object.entries(attrObj).forEach(([key, value]) => {```

File: index.js (hunk 5 above)
Comment (grandalf6) [reply index.js]: ```HOME_MENU_BUTTONS_ARRAY.forEach(({id, innerText}) => {```

**Review by grandalf6 — CHANGES_REQUESTED** [review]
Still some issues to resolve

File: index.css (hunk 8 above)
Comment (Varga4x4) [reply index.css]: Nope, solved.

File: index.js (hunk 7 above)
Comment (Varga4x4) [reply index.js]: True, removed it.

File: index.js (hunk 5 above)
Comment (Varga4x4) [reply index.js]: Solved

File: index.js (hunk 6 above)
Comment (Varga4x4) [reply index.js]: Solved

File: index.js [hunk 9]
```diff
@@ -29,18 +29,6 @@ const HOME_MENU_BUTTONS_ARRAY = [
     },
 ]
 
-const ELEMENT_IDS = {
```
Comment (grandalf6) [inline index.js]: You mentioned that you were going to use it in a future. I hope this is coming in some time.

File: index.js [hunk 10]
```diff
@@ -75,11 +61,11 @@ const renderHomeMenu = () => {
         innerText: "HOME"
     }, mainElement)
 
-    HOME_MENU_BUTTONS_ARRAY.forEach((option) => {
+    HOME_MENU_BUTTONS_ARRAY.forEach(({ id, innerText }) => {
         createElement("button", {
-            id: option.id,
+            id: id,
```
Comment (grandalf6) [inline index.js]: You could use object shorthand

File: index.js [hunk 11]
```diff
@@ -75,11 +61,11 @@ const renderHomeMenu = () => {
...
             className: "home-options",
-            innerText: option.innerText
+            innerText: innerText
```
Comment (grandalf6) [inline index.js]: Here as well

**Review by grandalf6 — CHANGES_REQUESTED** [review]
Also good

File: index.js (hunk 10 above)
Comment (Varga4x4) [reply index.js]: Solved

File: index.js (hunk 11 above)
Comment (Varga4x4) [reply index.js]: Solved

**Review by grandalf6 — APPROVED** [review]
Great

**Pull request merged**